binaries/coordinator/src/lib.rs

```diff
@@ -663,17 +658,17 @@ struct DaemonConnection {
 }
 
 async fn handle_destroy(
-    running_dataflows: &HashMap<Uuid, RunningDataflow>,
+    running_dataflows: &mut HashMap<Uuid, RunningDataflow>,
     daemon_connections: &mut HashMap<String, DaemonConnection>,
     abortable_events: &futures::stream::AbortHandle,
     daemon_events_tx: &mut Option<mpsc::Sender<Event>>,
     clock: &HLC,
 ) -> Result<(), eyre::ErrReport> {
     abortable_events.abort();
-    for (&uuid, dataflow) in running_dataflows {
-        stop_dataflow(
-            dataflow,
-            uuid,
+    for dataflow_uuid in running_dataflows.keys().cloned().collect::<Vec<_>>() {
+        let _ = stop_dataflow(
```

Comment:
Do we really need to change the stop_dataflow return type if we are not using it?

Reply:
We are using it in the control loop for `dora stop` but not for the `dora destroy` command. But its more readable if we don't have 2 different functions to do that.
```rust
let dataflow = stop_dataflow(
    &mut running_dataflows,
    dataflow_uuid,
    &mut daemon_connections,
    clock.new_timestamp(),
    grace_duration,
)
.await;

match dataflow {
    Ok(dataflow) => {
        dataflow.reply_senders.push(reply_sender);
    }
    Err(err) => {
        let _ = reply_sender.send(Err(err));
    }
}
```